### DISCUSSION TITLE
Force 16bit MotionT [export omf]

### DISCUSSION THREAD

**Discussion by LittleBabyGABE** [PavelBlend/blender-xray#457] [Q&A] [answered]
КАЖЕТСЯ сейчас такой возможности в аддоне нету, В базовом сдк это галочка есть и она оч полезная.
В самом X-Ray это флаг flTKey16IsBit, в аддоне KPF_T_HQ

Предыстория:
Мой давнешний воркфлоу:
-3dsmax2018 - создание анимации
-3dsmax2016 - корректировка некорректных ключей через рееэкспорт из этого макса
-ActorEditor - компановка всех skl анимаций и экспорт их в omf(с галочкой 16 бит).
С недавнего времени так вышло что захотел через блендер сократить себе процесс. В итоге убрал из воркфлоу 3dsmax 2016 и actoreditor и был невероятно рад. В один момент показалось что всё нормально и анимации экспортируются с флагом 16bit, но это не так. В блендере пока плохо разбираюсь.

Решил попытаться временно сделать самостоятельно правку для себя. Нашёл файлик omf\exp.py. В нём нашёл 2 места где упоминается формат экспорта(функции  get_motions и export_omf). 3b - 8 бит; 3h - 16 бит. Позаменял на 3h где казалось логичным, но в итоге лишь файл с бОльшим размером(ожидаемо), а в самой игре ломаются матрицы у всего кроме кости левой ключицы. l_clavicle это первая кость после корневой bip01, все остальные кости некорректно экспортируются и их полигоны не рендерятся. 

Было бы круто получить помощь с временным решением. Ну и внедрить в релиз такую галочку форса омф экспорта в 16 бит. :)

**Comment by PavelBlend** [comment, marked as answer]
@LittleBabyGABE добавил поддержку экспорта 16 битных анимаций. Скачать можно из этого коммита:
https://github.com/PavelBlend/blender-xray/tree/6a7e5f73ec5e105f2a96093668c8901b5e0239c6

Тестировал в игре, могу скинуть gamedata папку, чтобы ты смог быстро протестировать:
[test_omf_16_bit.zip](https://github.com/PavelBlend/blender-xray/files/7921303/test_omf_16_bit.zip)

Вроде бы ничего не сломал. Если будут ошибки, пиши.

В настройки экспорта была добавлена галочка:
![01](https://user-images.githubusercontent.com/7983249/150696312-cf4191f1-1cf2-46d5-8102-13d1f29432e0.jpg)
